packages/protocol/contracts/tokenvault/ERC1155Vault.sol

```diff
@@ -58,8 +59,8 @@ contract ERC1155Vault is BaseNFTVault, ERC1155ReceiverUpgradeable {
         onlyValidAmounts(opt.amounts, opt.tokenIds, false)
     {
         if (
-            !IERC165(opt.token).supportsInterface(ERC1155_INTERFACE_ID)
-                || IERC165(opt.token).supportsInterface(ERC721_INTERFACE_ID)
+            !IERC165Upgradeable(opt.token).supportsInterface(ERC1155_INTERFACE_ID)
```

Comment:
this is also buggy, we have to use try-catch for 167 interface testing, see comments above.

Reply:
Please search for `!IERC165Upgradeable` and fix all the problems by using try-catch.